[Parties] Abort party invitation creation if my current party is already fulll

Ikreb1 opened the pull request:
A simple check that stops you from creating more invites if your party is already full. Aborted with status code BAD_REQUEST.

judgeaxl commented:
You'll need to lock the party around the size check and the member add, otherwise another request may add a member in between.

Also, all other checks that aren't permissions checks happen in the create_party_invite function, so I think this logic should belong there, not in the request handler

Ikreb1 commented:
I moved the check and added a watch to **inviting_player_party_key**, but I feel like I need to add a watch to the **party_players_key** inside **get_party_members** for this to be correct

einarth commented on driftbase/tests/test_parties.py:
```diff
@@ -292,6 +292,34 @@ def test_leaving_party_invalidates_outstanding_invites_from_same_player(self):
+        for guest_name in [self.make_user_name(f"Guest_{i}") for i in range(4)]:
+            self.make_named_player(username=guest_name)
+            guest_ids.append(self.player_id)
+            guest_names.append(guest_name)
```
you can make this simpler by using make_player, i.e.
`for i in range(4):
    guest_names.append(self.make_player())
    guest_ids.append(self.player_id)
`

Ikreb1 commented:
@judgeaxl This should work now with the latest change :)
